src/login/logind-dbus.c

```diff
@@ -2038,6 +2044,94 @@ static usec_t nologin_timeout_usec(usec_t elapse) {
         return LESS_BY(elapse, 5 * USEC_PER_MINUTE);
 }
 
+int manager_load_scheduled_shutdown(Manager *m) {
+        _cleanup_fclose_ FILE *f = NULL;
+        _cleanup_free_ char *usec = NULL,
+               *warn_wall = NULL,
+               *mode = NULL,
+               *wall_message = NULL,
+               *uid = NULL,
+               *tty = NULL;
+        _cleanup_free_ char *unescaped = NULL;
+        int r;
+
+        assert(m);
+
+        r = parse_env_file(f, SHUTDOWN_SCHEDULE_FILE,
+                        "USEC", &usec,
+                        "WARN_WALL", &warn_wall,
+                        "MODE", &mode,
+                        "WALL_MESSAGE", &wall_message,
+                        "UID", &uid,
+                        "TTY", &tty);
+        if (r == -ENOENT)
+                return 0;
+        if (r < 0) {
+                log_debug_errno(r, "Failed to parse " SHUTDOWN_SCHEDULE_FILE ": %m");
+                goto fail;
+        }
+
+        r = deserialize_usec(strnull(usec), &m->scheduled_shutdown_timeout);
+        if (r < 0)
+                goto fail;
+
+        HandleAction handle = handle_action_from_string(mode);
+        if (handle == _HANDLE_ACTION_INVALID) {
+                log_debug_errno(SYNTHETIC_ERRNO(EINVAL), "Failed to parse scheduled shutdown type '%s'", mode);
```

Comment:
log_debug_errno() returns the (negative) error number you pass in (but strips the effect of SYNTHETIC_ERRNO(). Hence you can merge this line and the next into:

```c
r = log_debug_errno(SYNTHETIC_ERRNO(EINVAL), …);
```

Reply:
(but as mentioned, I think we should continue with parsing fields here, to handle failures of parsing our own internal serialization as gracefully as possible)